scripts/xtensa-build-zephyr.py

```diff
@@ -891,15 +881,6 @@ def build_platforms():
 	src_dest_list = []
 	tools_output_dir = pathlib.Path(STAGING_DIR, "tools")
 
-	# Install sof-logger from the last platform built (requires building at least one platform).
-	if platform_build_dir_name:
-		sof_logger_dir = pathlib.Path(west_top, platform_build_dir_name, "zephyr",
-					      "sof-logger_ep", "build", "logger")
-		sof_logger_executable_to_copy = pathlib.Path(shutil.which("sof-logger", path=sof_logger_dir))
-		sof_logger_installed_file = pathlib.Path(tools_output_dir, sof_logger_executable_to_copy.name).resolve()
-
-		src_dest_list += [(sof_logger_executable_to_copy, sof_logger_installed_file)]
```

Comment:
Honestly I'm hoping someone chimes in to clarify this.

AFAICT sof-logger, smex, .ldc files and old-style trace context usage are all degenerate in Zephyr builds.  The window that used to store the dictionary logging output has been repurposed to transmit plaintext Zephyr logs, and almost nothing is actually emitting the trace output anyway as all the {tr,comp}_{err,warn,info}() has been redirected to Zephyr LOG_xxx() macros which ignore the context struct.

Basically this is me ripping the bandaid off to see what breaks.  For sure, after this PR merges none of that stuff will work anymore as the addresses of the sof_uuid structs won't appear within the static_uuids section and sof-logger won't see any ldc entries.

If we absolutely still need this, let me know and I'll see what needs to be ported.

Reply:
@andyross @lyakh Let's have some from @thesofproject/nxp check this. In the transition to Zephyr, we did use sof-logger early with Zephyr so the smex/ldc/soflogger stuff was also used with Zephyr, but indeed for Intel platforms we've now moved on to Zephyr logging subsystem completely. But has NXP made the move? There is certainly a maintainance cost in having these around (see https://github.com/thesofproject/sof/pull/8822 ), so in general I'm happy to drop these.

PS The UUID rework looks great. Once we have the v2.10 release out, I'll do a more proper review.

Reply:
@thesofproject/mediatek should also check to see if they still depend on this

Reply:
FWIW: mt8xxx builds are fine in-tree as non-Zephyr builds are unchanged and will continue to work.

The question here is whether or not any Zephyr configurations are still in the deprecated/only-partially-working trace contexts.   And I'm pretty sure[1] the answer is no, because if you tried you'd discover that it doesn't actually have all the info.  When __ZEPHYR__==1, all the tr_info()/comp_info() logging et. al. goes to a Zephyr LOG_INFO() macro, 100% of the time.  And the implementation of that ignores the trace context that would have been used earlier to associate the log message with a UUID[2].  So the only output you'll get out of it are a handful of spots that are still going to the trace context (buffer and dai logging, in particular, are still using trace-based macros which land in stubs in a Zephyr build, though AFAICT those don't appear on the Zephyr output and no one seems to care!).

Anyway, the point being: this PR moves the UUIDs to a different location in the binary (an iterable array and not a special `.static_uuids` section) with a different layout (the `struct sof_uuid_entry` that includes the string name in addition to the bare UUID).  So it's 100% going to break any existing users.  I just don't think there are any.

[1] I mean, actually I'm more than pretty sure.  But software is complicated and edge cases abound, thus the questions.

[2] Zephyr logging is plaintext, ignores the trace context and UUID, and doesn't require .ldc files nor smex/sof_logger.  There actually is a Zephyr dictionary logging feature @dcpleung did long ago, but it's not enabled for SOF and to be honest I don't know much about how its tooling works.

Reply:
Oh i totally misread this as C code, not python, then yes ignore my previous comment since this is zephyr specific

Reply:
Unfortunately we still have old and very complex build and CI dependencies here because (as usual) the logging transitions to Zephyr and IPC4 were after-thoughts and rushed on the validation side. Some in Intel CI are not even open-source, which means you couldn't fix them even if you volunteered to.

So unless this literally **stops building**, I highly recommend you don't touch this if you want the rest merged in a reasonable time. Feel free to print additional warnings.

Reply:
@andyross As for Zephyr logging being plaintext, we have enabled both Zephyr dictionary and mipi syst and even have an overlay in upstream SOF -> work/sof/app/logging_mipisystcat_overlay.conf . Enablement is pending changes to CI infra which has a lot of dependencies to the current plaintext logs, but we have full capability to employ dictionaries.

Reply:
@andyross I'm good to go ahead without CI being ready - can you wrap any transition a a Kconfig which we can enables as CI catches up.

Reply:
CI should be fine, almost all of it was just waiting for the missing Zephyr patch mentioned above, which is in now that #9278 has merged.[1]

Should have it up today, July-4th-distractions willing.

[1] Also there was a tweak needed for testbench, which builds the SOF code in a third (!) cmake environment to make a shared library for tplg_parser which early-exits (!!) from the legacy CMakeLists.txt, so I needed to move where the generation was included.